### DISCUSSION TITLE
Missed reading alert does not work

### DISCUSSION THREAD

**Discussion by Massimaux** [NightscoutFoundation/xDrip#3717] [Bug reports]
Hi,

I have been running xDrip+ on **Xiaomi 11 Lite 5G NE** for more than 2 years as a follower. Up till recently everything seemed to work just fine. A couple of days ago I accidentally woke up in the middle of the night and realized that there had been no signal from my son's master for about an hour. Within that hour his blood sugar fell to 3.x mmol/L, remained hypo for 15-20 minutes and luckily increased to the 4+ safe zone by itself. I got upset because my son's glycemia went low while there was no signal from the master and the **MISSED READING ALERT set to 15 minutes didn't trigger to wake me up**.

Next day I installed and tried several different versions of xDrip+ (from 2024, 2023, 2019) and all of them failed to trigger the missed reading alert.

What seemed suspicious was the fact that after installing each xDrip+ version, I received this notification at the opening (see attached screenshots): 

_**This app was built for an older version of Android. It might not work properly and it doesn't include the latest security and privacy protections.**_

Then I checked my current version of Android. It is Xiaomi HyperOS. I guess it has been upgraded from MIUI recently.

Is the inability of xDrip+ to trigger the Missed Reading Alert related to the upgrade from MIUI to HyperOS? I'm 100% sure that it worked properly up till recently.

And, what should I do to make this extremely important alert function again?

I hope there is a simple solution for this unfortunate problem.

Thank you,
Marjan

![viber_image_2024-10-13_18-44-36-819](https://github.com/user-attachments/assets/87dc6481-a6c8-4bb8-87eb-3d5b5e3c2d27)
![viber_image_2024-10-13_18-44-36-849](https://github.com/user-attachments/assets/5d9d5e21-c220-479d-a809-b73f6a1abf02)

**Comment by Navid200** [comment]
Android is updated continuously.  We cannot abandon old versions of Android right away because worldwide, there are many who still use older versions.
In order for the older versions to run, there are a few minor tidbits some of us have to endure.  The note you have spotted is one of them.  it is just a nuisance and you can go ahead and install xDrip after seeing it.

But, let's talk about why missed reading alert did not work for you.
Would you please post a screenshot of your missed reading alert settings page?

**Comment by Massimaux** [comment]
Here is the missed reading alert settings page:

![Screenshot_2024-10-13-21-28-26-380_com eveningoutpost dexdrip-edit](https://github.com/user-attachments/assets/b70941d3-09c0-4e28-9e0a-f63f9628a75b)

**Reply by Navid200** [reply]
Can you scroll through the xDrip logs and find the time related to when the glucose was low and you had missed readings, but got no missed reading alert?

**Reply by Massimaux** [reply]
I'm sorry, I can't do that because I reinstalled the app several times since then and lost that log.

**Reply by Navid200** [reply]
I can only tell you what could have caused it with no way of proving it in that case.

1- The missed reading alert unfortunately does not use Volume profiles.  Therefore, if you lower the volume on your phone, you may never hear the missed reading alert.
This is something I hope we can change in near future.

2- You may have selected to only receive missed reading alerts during a time of day that was different than the time you had missed readings.

**Reply by Massimaux** [reply]
Regarding these possibilities:

1- The sound may have been missed but there must still be a notification text on the screen when missed reading alert occurs. The notification text doesn't show up either.

2-The alert is set for all day. No time restrictions.

**Reply by Navid200** [reply]
I am running xDrip 2024.10.11 on Android 15 beta.
I just put my phone in the microwave so that it will get no readings.

This is what the screen looks like:
![Screenshot_20241013-161300](https://github.com/user-attachments/assets/3bc1079f-e964-4a6a-a648-52ef420702f7)  
  
The additional drop means there is a notification.  
  
Pulling down from the top to expand the notifications, you can see this: 
![Screenshot_20241013-161320](https://github.com/user-attachments/assets/e6650a5e-3fe1-45f2-99bc-c38d7185e310)  
  
That is the missed reading alert notification.  
  
Looking at the logs, you can see this:  
![Screenshot_20241013-161341](https://github.com/user-attachments/assets/2265af34-bc82-46e8-a00d-3448c58bf126)  
  
The log at the top is the missed reading alert that went off.  
  
There is nothing else I can do to verify that everything is OK with xDrip.  
If you ever experience this again, please don't erase the logs and hopefully, we can narrow down what could be the cause.

**Reply by Massimaux** [reply]
I disabled wifi and mobile data to cause missed readings. No missed reading alert was generated after 15 minutes.

Here are the screenshots.
![Screenshot_2024-10-13-22-37-22-238_com eveningoutpost dexdrip-edit](https://github.com/user-attachments/assets/0741c014-d341-4d34-bab3-b9f08b064913)
![Screenshot_2024-10-13-22-38-03-938_com eveningoutpost dexdrip](https://github.com/user-attachments/assets/49546958-ae7c-49fa-ab4e-5b3446441727)

**Reply by Navid200** [reply]
@jamorham What could be causing this?

**Reply by Navid200** [reply]
Can you post a screenshot of the classic status page?

**Reply by Massimaux** [reply]
What is classic status page?

**Reply by Navid200** [reply]
This:  https://navid200.github.io/xDrip/docs/xDrip-Version.html

**Reply by Massimaux** [reply]
![Screenshot_2024-10-16-18-16-39-112_com eveningoutpost dexdrip-edit](https://github.com/user-attachments/assets/eff58d7f-3acf-4657-a057-ec7a402e3558)
